support/cas-server-support-duo-core/src/main/java/org/apereo/cas/adaptors/duo/authn/DuoAuthenticationHandler.java

```diff
@@ -124,4 +113,14 @@ public boolean supports(final Credential credential) {
         return DuoCredential.class.isAssignableFrom(credential.getClass())
             || credential instanceof DuoDirectCredential;
     }
+
+    /**
+     * Returns the id of the provider used for this authentication handler.
+     *
+     * @return - the provider id
+     */
+    public String getProviderId() {
```

Comment:
Unless you mean to make the method private of sorts, it would be better if you actually returned the provider instance, and renamed the method to reflect that bit too.